src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp

```diff
@@ -45,6 +50,26 @@ std::string getMetadataKeyPrefix(ObjectStoragePtr object_storage)
         : metadata_key_prefix;
 }
 
+void loadDirectoryTree(InMemoryDirectoryPathMap::Map & map, ObjectStoragePtr object_storage)
+{
+    const auto common_key_prefix = object_storage->getCommonKeyPrefix();
+    LOG_DEBUG(getLogger("MetadataStorageFromPlainObjectStorage"), "Loading directory tree");
+    for (auto & [local_path, info] : map)
```

Comment:
Can be parallelized with `ThreadPoolCallbackRunnerLocal`;
I doesn't work well for the hierarchical structure because nested directories are traversed several times.